ruma-events-macros/src/event_enum.rs

```diff
@@ -1,126 +1,64 @@
 //! Implementation of event enum and event content enum macros.
 
+use matches::matches;
 use proc_macro2::{Span, TokenStream};
-use quote::quote;
+use quote::{format_ident, quote};
 use syn::{
     parse::{self, Parse, ParseStream},
     Attribute, Expr, ExprLit, Ident, Lit, LitStr, Token,
 };
 
-use crate::event_names::{
-    ANY_BASIC_EVENT, ANY_EPHEMERAL_EVENT, ANY_MESSAGE_EVENT, ANY_STATE_EVENT,
-    ANY_STRIPPED_STATE_EVENT, ANY_SYNC_MESSAGE_EVENT, ANY_SYNC_STATE_EVENT, ANY_TO_DEVICE_EVENT,
-    REDACTED_MESSAGE_EVENT, REDACTED_STATE_EVENT, REDACTED_STRIPPED_STATE_EVENT,
-    REDACTED_SYNC_MESSAGE_EVENT, REDACTED_SYNC_STATE_EVENT,
-};
-
-// Arrays of event enum names grouped by a field they share in common.
-const ROOM_EVENT_KIND: &[&str] = &[
-    ANY_MESSAGE_EVENT,
-    ANY_SYNC_MESSAGE_EVENT,
-    ANY_STATE_EVENT,
-    ANY_SYNC_STATE_EVENT,
-    REDACTED_MESSAGE_EVENT,
-    REDACTED_STATE_EVENT,
-    REDACTED_SYNC_MESSAGE_EVENT,
-    REDACTED_SYNC_STATE_EVENT,
-];
-
-const ROOM_ID_KIND: &[&str] = &[
-    ANY_MESSAGE_EVENT,
-    ANY_STATE_EVENT,
-    ANY_EPHEMERAL_EVENT,
-    REDACTED_STATE_EVENT,
-    REDACTED_MESSAGE_EVENT,
-];
-
-const EVENT_ID_KIND: &[&str] = &[
-    ANY_MESSAGE_EVENT,
-    ANY_SYNC_MESSAGE_EVENT,
-    ANY_STATE_EVENT,
-    ANY_SYNC_STATE_EVENT,
-    REDACTED_SYNC_STATE_EVENT,
-    REDACTED_SYNC_MESSAGE_EVENT,
-    REDACTED_STATE_EVENT,
-    REDACTED_MESSAGE_EVENT,
-];
-
-const SENDER_KIND: &[&str] = &[
-    ANY_MESSAGE_EVENT,
-    ANY_STATE_EVENT,
-    ANY_SYNC_STATE_EVENT,
-    ANY_TO_DEVICE_EVENT,
-    ANY_SYNC_MESSAGE_EVENT,
-    ANY_STRIPPED_STATE_EVENT,
-    REDACTED_MESSAGE_EVENT,
-    REDACTED_STATE_EVENT,
-    REDACTED_STRIPPED_STATE_EVENT,
-    REDACTED_SYNC_MESSAGE_EVENT,
-    REDACTED_SYNC_STATE_EVENT,
-];
+fn is_non_stripped_room_event(kind: &EventKind, var: &EventKindVariation) -> bool {
+    matches!(kind, EventKind::Message(_) | EventKind::State(_))
+        && !matches!(var, EventKindVariation::Stripped | EventKindVariation::RedactedStripped)
+}
 
-const PREV_CONTENT_KIND: &[&str] = &[ANY_STATE_EVENT, ANY_SYNC_STATE_EVENT];
+fn has_room_id_field(kind: &EventKind, var: &EventKindVariation) -> bool {
+    matches!(kind, EventKind::Message(_) | EventKind::State(_))
+        && matches!(var, EventKindVariation::Full | EventKindVariation::Redacted)
+}
```

Comment:
This is only used once so can be inlined, no?

Reply:
That is a much better metric than size and how it ends up being formatted :laughing: oops.